.github/workflows/pull-request.yml

```diff
@@ -443,14 +440,22 @@ jobs:
         run: npm run generate-docs
 
       - name: Commit and Push Changes
-        run: |
-          git config user.name "github-actions[bot]"
-          git config user.email "${{ github.event.pull_request.user.id }}+${{ github.event.pull_request.user.login }}@users.noreply.github.com"          git add .
-          git commit -m "Auto-generate documentation"
-          git push origin HEAD:${{ github.head_ref }}
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
+          HEAD_REF: ${{ github.head_ref }}
+        run: |
+          if [[ $(git status --porcelain) ]]; then
+            echo "Changes detected. Auto-committing updates."
+            git config user.name "github-actions[bot]"
+            git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+            git add .
```

Comment:
You need a `git add --all` command too. This will cover new files created due to future code modifications